app/controllers/vms_controller.rb

```diff
@@ -55,6 +57,8 @@ def update
     @vm.sudo_users = info_params[:sudo_user_ids]
     @vm.users = info_params[:non_sudo_user_ids]
     @vm.ensure_config.description = info_params[:description]
+    @ip = @vm.ensure_config.ip = info_params[:ip]
+    @dns = @vm.ensure_config.dns = info_params[:dns]
```

Comment:
Here we can conform to rails convention and use @vm.ensure_config.update(config_params)